packages/bitcoin-js-wallet-provider/lib/BitcoinJsWalletProvider.js

```diff
@@ -1,48 +1,34 @@
-import Provider from '@liquality/provider'
-import { AddressTypes, selectCoins } from '@liquality/bitcoin-utils'
+import BitcoinWalletProvider from '@liquality/bitcoin-wallet-provider'
+import WalletProvider from '@liquality/wallet-provider'
 import * as bitcoin from 'bitcoinjs-lib'
 import * as bitcoinMessage from 'bitcoinjs-message'
-import { Address, addressToString } from '@liquality/utils'
+import { addressToString } from '@liquality/utils'
 import { mnemonicToSeed } from 'bip39'
-import { fromSeed } from 'bip32'
-import { BigNumber } from 'bignumber.js'
+import bip32 from 'bip32'
 
 import { version } from '../package.json'
 
-const ADDRESS_GAP = 20
-const NONCHANGE_ADDRESS = 0
-const CHANGE_ADDRESS = 1
-const NONCHANGE_OR_CHANGE_ADDRESS = 2
-
-const ADDRESS_TYPE_TO_LEDGER_PREFIX = {
-  'legacy': 44,
-  'p2sh-segwit': 49,
-  'bech32': 84
-}
-
-export default class BitcoinJsWalletProvider extends Provider {
+export default class BitcoinJsWalletProvider extends BitcoinWalletProvider(WalletProvider) {
   constructor (network, mnemonic, addressType = 'bech32') {
-    super()
-    if (!AddressTypes.includes(addressType)) {
-      throw new Error(`addressType must be one of ${AddressTypes.join(',')}`)
-    }
+    super(network, addressType, [network])
     if (mnemonic === '') {
```

Comment:
Can we instead check for truthiness of `mnemonic`?

```
if (mnemonic) {}
```

Reply:
fixed